### PR TITLE
[Flang2] Support for noinline/forceinline pragmas

### PR DESCRIPTION
**Pull request by Chirag-Khandelwal**
Apologies for the new PR, due to my misunderstanding, I deleted the repository forks from which the original PRs were created (closed the older PRs).

@kiranchandramohan This PR fixes the issue of attribute not coming up with `-g` option as well.

### CODE REVIEW

File: tools/shared/pragma.c
Comment (SouraVX) [inline tools/shared/pragma.c:1308]: NIT: This seems ambiguous(2 statements following `else` without explicit braces) ?
WDYT?

File: tools/flang2/flang2exe/cgmain.cpp
Comment (SouraVX) [inline tools/flang2/flang2exe/cgmain.cpp:13263]: I assuming this is already documented ? this `NOINLINE` bit you're using :)

File: tools/flang2/flang2exe/cgmain.cpp
Comment (SouraVX) [inline tools/flang2/flang2exe/cgmain.cpp:13271]: NIT: there's seems to be an extra space following `alwaysinline`. Is it intended or TYPO :)  ?

File: tools/shared/pragma.c
Comment (pawosm-arm) [reply tools/shared/pragma.c:1308]: Definitely needs to be addressed.

File: tools/flang2/flang2exe/cgmain.cpp
Comment (Chirag-Khandelwal) [reply tools/flang2/flang2exe/cgmain.cpp:13271]: Yea, that's intended.

File: tools/flang2/flang2exe/cgmain.cpp
Comment (Chirag-Khandelwal) [reply tools/flang2/flang2exe/cgmain.cpp:13263]: Yes it is
> Don't extract this function (set by pragma noinline)

:)

File: tools/shared/pragma.c
Comment (Chirag-Khandelwal) [reply tools/shared/pragma.c:1308]: That's a mistake on my part. Apologies, thanks for finding it.
Fixed.

**Review by pawosm-arm — APPROVED** [review]

File: tools/shared/pragma.c
Comment (bryanpkc) [inline tools/shared/pragma.c:434]: Quoting myself from the old pull request:

> I don't think you need to support noforceinline since another pragma named noinline exists for exactly the same purpose. Also, your test cases did not cover the use of noforceinline.

What I meant to suggest was that you could change the `true` to `false` on this line.

Comment (bryanpkc) [reply tools/shared/pragma.c:434]: The `noforceinline` test would be unnecessary.

Comment (Chirag-Khandelwal) [reply tools/shared/pragma.c]: Oh apologies, I missed that.
Updated the code.

**Review by bryanpkc — APPROVED** [review]
LGTM.

File: test/directives/dir_forceinline.f90
Comment (bryanpkc) [inline test/directives/dir_forceinline.f90:21]: Nit: Testing for an unsupported pragma is unnecessary.

Comment (Chirag-Khandelwal) [reply test/directives/dir_forceinline.f90]: Removed that test.

**Review by SouraVX — APPROVED** [review]

**Comment by SouraVX** [comment]
If everyone's on-board then I can ask @shivaramaarao to merge this or I'm also Okay if anyone here having merge access  would like to merge this :)

**Pull request merged**
